flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/codegen/JsonGenerateUtils.scala

```diff
@@ -28,24 +28,42 @@ import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.databind.util.RawV
 
 import org.apache.calcite.rex.{RexCall, RexNode}
 
+import java.time.format.DateTimeFormatter
+
 /** Utility for generating JSON function calls. */
 object JsonGenerateUtils {
 
   /**
    * Returns a term which wraps the given <code>valueExpr</code> into a [[JsonNode]] of the
    * appropriate type.
    *
+   * @param ctx Context object.
    * @param containerNodeTerm Name of the [[ContainerNode]] from which to create the node.
    * @param valueExpr Generated expression of the value which should be wrapped.
    * @return Generated code fragment creating the appropriate node.
    */
-  def createPrimitiveNodeTerm(containerNodeTerm: String, valueExpr: GeneratedExpression): String = {
+  def createPrimitiveNodeTerm(
+      ctx: CodeGeneratorContext,
+      containerNodeTerm: String,
+      valueExpr: GeneratedExpression): String = {
     valueExpr.resultType.getTypeRoot match {
       case CHAR | VARCHAR => s"$containerNodeTerm.textNode(${valueExpr.resultTerm}.toString())"
       case BOOLEAN => s"$containerNodeTerm.booleanNode(${valueExpr.resultTerm})"
       case DECIMAL => s"$containerNodeTerm.numberNode(${valueExpr.resultTerm}.toBigDecimal())"
       case TINYINT | SMALLINT | INTEGER | BIGINT | FLOAT | DOUBLE =>
         s"$containerNodeTerm.numberNode(${valueExpr.resultTerm})"
+      case TIMESTAMP_WITHOUT_TIME_ZONE =>
+        val formatter = s"${typeTerm(classOf[DateTimeFormatter])}.ISO_LOCAL_DATE_TIME"
+        s"$containerNodeTerm.textNode(${valueExpr.resultTerm}.toLocalDateTime().format($formatter))"
```

Comment:
fyi: with this we have different timestamp semantics in our formats vs functions.
https://ci.apache.org/projects/flink/flink-docs-master/docs/connectors/table/formats/json/#json-timestamp-format-standard